lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1171,6 +1173,112 @@ doesAccessorNeedDynamicAttribute(AccessorDecl *accessor, Evaluator &evaluator) {
   }
 }
 
+llvm::Expected<bool>
+IsFinalRequest::evaluate(Evaluator &evaluator, ValueDecl *decl) const {
+  switch (decl->getKind()) {
+    case DeclKind::Var: {
+      // Properties are final if they are in a static class or a 'let'
```

Comment:
`s/in a static class/declared 'static'/`

The `static` keyword, inside a class, is a synonym for `class final`. `class` means something is not an instance member so you call it on an instance of the metatype and not an instance of the type.